### PR TITLE
MySQL configuration

### PR DESCRIPTION
**Pull request by brianyu28**
This PR just creates MySQL tables if they don't already exist upon program launch; it doesn't actually insert anything into the tables just yet (@annieechen is working on that, and will submit a separate PR for that after this one is merged).

### CODE REVIEW

File: model.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import MySQLdb as mdb
+import os
+from warnings import filterwarnings
+
+# establish connection to database
+con = mdb.connect(
+    os.environ['MYSQL_HOST'],
+    os.environ['MYSQL_USER'],
```
Comment (dmalan) [inline model.py]: Sorry, should be `_USERNAME` and `_PASSWORD` now.

File: model.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    os.environ['MYSQL_PASS'],
+    os.environ['MYSQL_DATABASE'],
+    charset='utf8'
+)
+cur = con.cursor()
+
+# disable 'table already exists' warnings
+filterwarnings('ignore', category=mdb.Warning)
```
Comment (dmalan) [inline model.py]: Just to recap, is this necessary, given that your SQL query already has `IF NOT EXISTS`?

File: model.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        input_id BIGINT,
+        output TEXT,
+        PRIMARY KEY (id),
+        FOREIGN KEY (input_id) REFERENCES inputs(id) ON DELETE CASCADE
+    ) ENGINE=InnoDB;
+    '''
+    
+    cur.execute(inputs_schema)
+    cur.execute(outputs_schema)
+
```
Comment (dmalan) [inline model.py]: blank line

File: model.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    
+    outputs_schema = '''
+    CREATE TABLE IF NOT EXISTS outputs (
+        id BIGINT NOT NULL AUTO_INCREMENT,
+        input_id BIGINT,
+        output TEXT,
+        PRIMARY KEY (id),
+        FOREIGN KEY (input_id) REFERENCES inputs(id) ON DELETE CASCADE
+    ) ENGINE=InnoDB;
+    '''
```
Comment (dmalan) [inline model.py]: Is this the best approach or is there a de facto standard (or at least popular) "migration" library we should be using, in anticipation of making future table changes?

File: model.py (hunk 2 above)
Comment (brianyu28) [inline model.py]: For some reason, it seems to be; even with `IF NOT EXISTS` present in the code, `MySQLdb` still prints out `table already exists` warnings every time `configure()` is executed unless the warnings are suppressed.

Comment (dmalan) [inline model.py]: k

**Comment by dmalan** [comment]
I've created the server-side DB and credentials, fyi. But will wait before merging (re question about migration).

File: model.py (hunk 4 above)
Comment (brianyu28) [inline model.py]: Though I haven't tried it before, looks like SQLAlchemy (http://www.sqlalchemy.org/) is a fairly popular library. It doesn't seem to support migrations natively, but there do appear to be extensions of it (https://pypi.python.org/pypi/alembic) which do.

Comment (dmalan) [inline model.py]: Hm, okay, want to poke around a bit more to see what most folks tend to use? Use case is... if we make changes to the DB tables down the road, ideally we just want the server to run a "migration," essentially a set of `ALTER` commands that updates the schema automatically.

File: model.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+# disable 'table already exists' warnings
+filterwarnings('ignore', category=mdb.Warning)
+
+def configure():
+    inputs_schema = '''
+    CREATE TABLE IF NOT EXISTS inputs (
+        id BIGINT NOT NULL AUTO_INCREMENT,
+        cmd VARCHAR(1024) NULL,
+        script TEXT NOT NULL,
+        username VARCHAR(32) NULL,
```
Comment (dmalan) [inline model.py]: PS let's add an `INDEX` on `username` and on `created` so we can implement hearts more efficiently (by querying for users' past usage).

**Comment by dmalan** [comment]
Just let me know when ready, holding off on merging for now.

**Comment by brianyu28** [comment]
Closing because using a migration library instead in PR #14

**Pull request closed without merging**